### DISCUSSION TITLE
GCC version for Allegro5 builds?

### DISCUSSION THREAD

**Discussion by Axle-Ozz-i-sofT** [liballeg/allegro5#1541] [General]
Hi team
I have just started using/learning Allegro. And threw Nuklear into the mix as well to get myself up to speed with both.
I am currently using "allegro-x86_64-w64-mingw32-gcc-13.2.0-posix-seh-dynamic-5.2.9.0.zip" binaries.
And building my own apps on an older "TDM-GCC 9.2.0".
It turns out that the TDM tool-chain doesn't have to GNU DLLs for the Allegro DLLs above, especially "libgcc_s_seh-1.dll".

Allegro V5.2.9.1 Depends: (GCC Version ?)
- libgcc_s_seh-1.dll
- libstdc++-6.dll
- libwinpthread-1.dll

I had a look through Winlibs which appears to contain the correct RT files.
**What toolchain and GCC version did you build Allegro 5.2.9.0[1] with?**
Just so that I can keep some level of compatibility with the above 3 dlls, as well as using the same GCC version for the static binaries "allegro-x86_64-w64-mingw32-gcc-13.2.0-posix-seh-static-5.2.9.0.zip"

TIA
Axle

P.S. I tried to create an account with "allegro.cc/forums/" and new accounts don't appear to work, so is it acceptable to ask questions here or post ideas etc?
For example I just created a Visual Style Editor for Nuklear using the Allegro5 rendering. Others may have interest in the source if they want Nuclear widgets for Allegro games :)
I am not aiming at any high end game releases so I am likely to run with Allegro for some general graphics and GUI apps. It just keeps it simple across Win and Linux. For example I took the source from Nuklear Visual style Editor and compiled it on Ubuntu 20.04 without any change to the code :)

P.S.S I also need to ask some questions (double check) on correctly adding the Allegro LICENCE agreement when distributing Allegro DLLS.

**Comment by SiegeLord** [comment]
The binaries are created in MSYS2, the gcc version + runtime are specified in the filename.

> I tried to create an account with "allegro.cc/forums/" and new accounts don't appear to work, so is it acceptable to ask questions here or post ideas etc?

Yep, it's fine. There's also discord.

**Reply by Axle-Ozz-i-sofT** [reply]
Hi, thanks [SiegeLord](https://github.com/SiegeLord) :)

> "gcc version + runtime are specified in the filename" 

Yeah, I had a bit of a lala moment there, but I think I was looking for the library build ie Winlibs or something else.
Anyways, I don't use MSYS2 but "winlibs-x86_64-posix-seh-gcc-13.2.0-mingw-w64msvcrt-11.0.1-r5.7z" should be OK.
>
And with the Allegro LICENCE.txt, is it OK to just place a copy of the licence with the DLLs as ALLEGRO.TXT or merge the different library licences under a single LICENCE file? Sorry, I am a bit green when it comes to releasing binaries and don't want to step on any toes :)

Best regards
Axle
